### PR TITLE
1st version of PiHole Pack

### PR DESCRIPTION
**Pull request by Hruuttila**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Description
New integration with opensource PiHole product (https://pi-hole.net/)

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [X] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Demisto Partner?
- Harri Ruuttila / SA

### CODE REVIEW

File: Packs/PiHole/Integrations/PiHole/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+## [Unreleased]
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/CHANGELOG.md:1]: please remove changelog file, its not needed

File: Packs/PiHole/Integrations/PiHole/PiHole.yml [hunk 2]
```diff
@@ -0,0 +1,220 @@
+commonfields:
+  id: PiHole
+  version: -1
+name: PiHole
+display: PiHole
+category: IT Services
+description: This is an integration to PiHole
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.yml:7]: please improve description to describe what this integration can be used for

File: Packs/PiHole/Integrations/PiHole/PiHole.yml [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+configuration:
+- display: Server URL (e.g. https://example.net)
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.yml:9]: please update the example

File: Packs/PiHole/Integrations/PiHole/PiHole.yml [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+  additionalinfo: Input the base url excluding the /admin/api.php part
+- display: Trust any certificate (not secure)
+  name: insecure
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  type: 8
+  required: false
+- display: Auth Token
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.yml:22]: please move this param to be below the url

File: Packs/PiHole/Integrations/PiHole/PiHole.yml [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+script:
+  script: ''
+  type: python
+  commands:
+  - name: pihole-get-version
+    arguments: []
+    outputs:
+    - contextPath: PiHole.Version.version
+      description: Version info
+      type: Unknown
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.yml:36]: for all context outputs, please set type

File: Packs/PiHole/pack_metadata.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "PiHole",
+    "description": "PiHole",
+    "support": "developer",
```
Comment (Itay4) [inline Packs/PiHole/pack_metadata.json:4]:
```suggestion
    "support": "xsoar",
```

File: Packs/PiHole/pack_metadata.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    "currentVersion": "1.0.0",
+    "author": "Harri Ruuttila",
```
Comment (Itay4) [inline Packs/PiHole/pack_metadata.json:6]:
```suggestion
    "author": "Cortex XSOAR",
```

File: Packs/PiHole/pack_metadata.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    "url": "",
```
Comment (Itay4) [inline Packs/PiHole/pack_metadata.json:7]:
```suggestion
    "url": "https://www.paloaltonetworks.com/cortex",
```

File: Packs/PiHole/pack_metadata.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    "email": "harri@ruuttila.com",
```
Comment (Itay4) [inline Packs/PiHole/pack_metadata.json:8]:
```suggestion
    "email": "",
```

File: Packs/PiHole/Integrations/PiHole/PiHole.py [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+
+class Client:
+    def __init__(self, base_url, verify, proxies, auth):
+        self.base_url = base_url
+        self.verify = verify
+        self.proxies = proxies
+        self.auth = '&auth=' + auth
+
+    def http_request(self, method, url_suffix):
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.py:22]: i suggest you use `BaseClient._http_request` instead

File: Packs/PiHole/Integrations/PiHole/PiHole.py [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+        try:
+            return res.json()
+        except ValueError:
+            data = {'Info': res.text}
+            return data
+
+
+def results_return(command, thingtoreturn):
+    finaldata = {'PiHole': {command: thingtoreturn}}
+    return demisto.results({
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.py:39]: please use `CommandResults` instead - https://xsoar.pan.dev/docs/integrations/code-conventions#commandresults

File: Packs/PiHole/Integrations/PiHole/PiHole.py [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    })
+
+
+def test_module(client):
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    result = client.http_request('GET', '')
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.py:55]: what does that request do ?

File: Packs/PiHole/Integrations/PiHole/PiHole.py [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+
+''' IMPORTS '''
+
+import requests
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' CONSTANTS '''
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.py:12]: this is not used

File: Packs/PiHole/Integrations/PiHole/PiHole.yml [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+    arguments: []
+    outputs:
+    - contextPath: PiHole.OverTimeData10mins
+      description: Data over last 10mins
+      type: Unknown
+    description: Data needed for generating the domains/ads over time graph on the
+      Pi-hole web dashboard
+  - name: pihole-get-topitems
+    arguments:
+    - name: entries
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.yml:62]:
```suggestion
    - name: limit
```

File: Packs/PiHole/Integrations/PiHole/PiHole.yml [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+    outputs:
+    - contextPath: PiHole.Status
+      description: Status
+      type: Unknown
+    description: Show status of pihole action (enabled - disabled)
+  - name: pihole-enable
+    arguments: []
+    outputs:
+    - contextPath: PiHole.Enable
+      description: 'Enabled '
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/PiHole.yml:112]: please improve description

File: Packs/PiHole/TestPlaybooks/PiHole-Test.yml [hunk 16]
```diff
@@ -0,0 +1,1166 @@
+id: PiHole-Test
+name: PiHole-Test
+version: -1
+fromversion: 4.5.0
```
Comment (Itay4) [inline Packs/PiHole/TestPlaybooks/PiHole-Test.yml:4]: remove the from version here

File: Packs/PiHole/Integrations/PiHole/CHANGELOG.md (hunk 1 above)
Comment (Hruuttila) [reply Packs/PiHole/Integrations/PiHole/CHANGELOG.md:1]: Deleted as requested.

File: Packs/PiHole/Integrations/PiHole/PiHole.yml (hunk 2 above)
Comment (Hruuttila) [reply Packs/PiHole/Integrations/PiHole/PiHole.yml:7]: Added more details into the description

File: Packs/PiHole/Integrations/PiHole/PiHole.yml (hunk 4 above)
Comment (Hruuttila) [reply Packs/PiHole/Integrations/PiHole/PiHole.yml:22]: Done

File: Packs/PiHole/Integrations/PiHole/PiHole.yml (hunk 5 above)
Comment (Hruuttila) [reply Packs/PiHole/Integrations/PiHole/PiHole.yml:36]: done

File: Packs/PiHole/Integrations/PiHole/PiHole.py (hunk 12 above)
Comment (Hruuttila) [reply Packs/PiHole/Integrations/PiHole/PiHole.py:55]: It just get the base url of the api to see it responds correctly.

File: Packs/PiHole/Integrations/PiHole/PiHole.py (hunk 10 above)
Comment (Hruuttila) [reply Packs/PiHole/Integrations/PiHole/PiHole.py:22]: I would rather not. The coding conventions is following previously advised conventions.

File: Packs/PiHole/Integrations/PiHole/PiHole.py (hunk 11 above)
Comment (Hruuttila) [reply Packs/PiHole/Integrations/PiHole/PiHole.py:39]: done

**Comment by Hruuttila** [comment]
Working on the README file.

**Comment by Hruuttila** [comment]
Added Readme file.

File: Packs/PiHole/pack_metadata.json [hunk 17]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "PiHole",
+    "description": "Integration with PiHole",
+    "support": "xsoar",
```
Comment (Itay4) [inline Packs/PiHole/pack_metadata.json:4]:
```suggestion
    "support": "community",
```

File: Packs/PiHole/pack_metadata.json [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+    "currentVersion": "1.0.0",
+    "author": "Cortex XSOAR",
```
Comment (Itay4) [inline Packs/PiHole/pack_metadata.json:6]: please change to your name here

File: Packs/PiHole/pack_metadata.json [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+    "url": "https://www.paloaltonetworks.com/cortex",
```
Comment (Itay4) [inline Packs/PiHole/pack_metadata.json:7]: please remove url and add your email

File: Packs/PiHole/Integrations/PiHole/README.md [hunk 20]
```diff
@@ -0,0 +1,1150 @@
+Pi-hole is a network-level advertisement and Internet tracker blocking application which acts as a DNS sinkhole and optionally a DHCP server, intended for use on a private network.
+This integration was integrated and tested with version xx of PiHole
```
Comment (Itay4) [inline Packs/PiHole/Integrations/PiHole/README.md:2]: can you please update the version here?

**Review by Itay4 — APPROVED** [review]

**Pull request merged**